### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by m-kuhn**
Two small changes to fix API docs and prefer https over plain http.

Need backporting

### CODE REVIEW

File: src/core/geometry/qgsgeometry.h [hunk 1]
```diff
@@ -175,7 +175,6 @@ class CORE_EXPORT QgsGeometry
      * \note In QGIS 2.x this method was named geometry().
      *
      * \since QGIS 3.0
-     * \see primitive()
```
Comment (nyalldawson) [inline src/core/geometry/qgsgeometry.h]: Oops, should be \see get()

Comment (m-kuhn) [reply src/core/geometry/qgsgeometry.h]: thanks, done

**Comment by nyalldawson** [comment]
Change looks good, but I'm curious why sipify didn't pick it up. Maybe the sipify test is broken at the moment?

**Comment by m-kuhn** [comment]
Good question, @3nids any idea?

**Comment by 3nids** [comment]
https://github.com/qgis/QGIS/pull/6514

**Comment by 3nids** [comment]
this PR is now failing :) 
Do I get a free beer?

**Comment by m-kuhn** [comment]
>  Do I get a free beer?

Yes :beer:

**Pull request merged**